src/wasm/wasm-binary.cpp

```diff
@@ -242,7 +247,7 @@ void WasmBinaryWriter::writeTypes() {
   BYN_TRACE("== writeTypes\n");
   auto start = startSection(BinaryConsts::Section::Type);
   o << U32LEB(numGroups);
-  if (getTypeSystem() == TypeSystem::Nominal) {
+  if (typeSystem == TypeSystem::Nominal) {
     // The nominal recursion group contains every type.
     o << S32LEB(BinaryConsts::EncodedType::Rec)
       << U32LEB(indexedTypes.types.size());
```

Comment:
For `--nominal`, it all works out because the rec groups all have size 1. For `--hybrid` it all works because the MVP function types will also have rec group size 1, unless the input had rec groups. I don't think we have any validation that rec groups are only allowed when GC is enabled, but that seems like a separate problem.